### PR TITLE
Add redo undo command

### LINKED ISSUE #123: Add Redo Command

### PR DESCRIPTION
**Pull request by Goh-Li-Ting**
Add commands to redo and undo commands.
Redo and undo make use of `VersionedAddressBook` which extends `AddressBook` with an undo/redo history.

### CODE REVIEW

File: src/main/java/seedu/address/model/VersionedAddressBook.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        // state check
+        return super.equals(otherVersionedAddressBook)
+                && addressBookStateList.equals(otherVersionedAddressBook.addressBookStateList)
+                && currentStatePointer == otherVersionedAddressBook.currentStatePointer;
+    }
+
+    /**
+     * Thrown when trying to {@code undo()} but can't.
+     */
+    public static class NoUndoableStateException extends RuntimeException {
```
Comment (Weiennn) [inline src/main/java/seedu/address/model/VersionedAddressBook.java:98]: is there a reason the exceptions is inside the `VersionedAddressBook` class itself and not in another folder like the rest of the exceptions?
If so, can you can include this under design considerations for the DG as well

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -45,6 +45,7 @@ public SortCommand(Comparator<Person> comparator) {
     public CommandResult execute(Model model, CommandHistory history) {
         requireNonNull(model);
         model.updateSortedPersonList(comparator);
+        model.commitAddressBook();
```
Comment (Weiennn) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:48]: if the `SortCommand` is committed to `VersionedAddressBook` should the `FindCommand` be as well?

**Review by Weiennn — APPROVED** [review]
LGTM

**Review by aexolate — APPROVED** [review]
LGTM.

**Pull request merged**